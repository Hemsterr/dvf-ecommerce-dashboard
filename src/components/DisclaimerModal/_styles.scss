.disclaimer {
  max-width: pxToRem(360px);
  text-align: center;
  margin: 0 auto;

  &__btnWrapper {
    display: flex;
    justify-content: center;
  }

  &__btn {
    background-color: $red;
    display: block;
    text-align: center;
    width: pxToRem(200px);
    margin-top: pxToRem(15px);
  }
}

@media only screen and (max-width: $screen-xs) {
  .disclaimer  {
    padding: 0 pxToRem(20px);
  }
}
