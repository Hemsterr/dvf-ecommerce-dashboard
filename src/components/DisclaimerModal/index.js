// @flow
// Libs
import React from 'react'

// Components
import Modal from '../Modal'
import Button from '../Button'

type Props = {
  handleCloseModal: Function,
  handleAcceptDisclaimer: Function,
}

class DisclaimerModal extends React.PureComponent<Props> {
  static defaultProps = {
    handleCloseModal: () => {},
    handleAcceptDisclaimer: () => {},
  }

  render() {
    const { handleCloseModal, handleAcceptDisclaimer } = this.props

    return (
      <Modal className="shipping-modal" handleCloseModal={handleCloseModal}>
        <div className="shipping-modal">
          <div className="disclaimer">
            <p className="shipping-modal__title">Disclaimer</p>
            <p className="shipping-modal__title">
              The price generated is a general quote.
            </p>
            <p className="shipping-modal__title">
              You will receive an email confirmation with the final price after
              the Hemster team receives and reviews your garments and alteration
              requests.
            </p>
            <Button
              className="btn btn__basic shipping-modal__btn"
              label="Accept"
              type="success"
              size="small"
              handleOnClick={() => handleAcceptDisclaimer()}
            />
            <div className="disclaimer__btnWrapper">
              <Button
                className="btn btn__basic disclaimer__btn"
                label="Decline"
                size="small"
                handleOnClick={() => handleCloseModal()}
              />
            </div>
          </div>
        </div>
      </Modal>
    )
  }
}

export default DisclaimerModal
