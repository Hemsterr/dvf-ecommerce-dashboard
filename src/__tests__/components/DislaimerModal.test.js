// Libs
import React from 'react'
import { render, fireEvent } from '@testing-library/react'

// Component
import DisclaimerModal from '../../components/DisclaimerModal'

const props = {
  handleCloseModal: jest.fn(),
  handleAcceptDisclaimer: jest.fn(),
}

const wrapper = (props = {}) => {
  const component = render(<DisclaimerModal {...props} />)

  return component
}

describe('DisclaimerModal component', () => {
  // Snapshot
  test('should render correctly DisclaimerModal component default props', () => {
    const { container } = wrapper(props)
    expect(container).toMatchSnapshot()
    expect(DisclaimerModal.defaultProps.handleCloseModal()).toBeUndefined()
    expect(
      DisclaimerModal.defaultProps.handleAcceptDisclaimer()
    ).toBeUndefined()
  })

  test('Render the button with props', () => {
    const { container, rerender } = wrapper(props)

    // Checking event click button
    const buttons = container.querySelectorAll('.btn')
    fireEvent.click(buttons[0])
    fireEvent.click(buttons[1])
    rerender(<DisclaimerModal {...props} />)
    expect(props.handleCloseModal).toBeTruthy()
    expect(props.handleAcceptDisclaimer).toBeTruthy()
  })
})
