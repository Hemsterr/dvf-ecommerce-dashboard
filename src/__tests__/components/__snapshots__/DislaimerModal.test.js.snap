// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisclaimerModal component should render correctly DisclaimerModal component default props 1`] = `
<div>
  <div
    class="modal-wrapper"
  >
    <div
      class="modal shipping-modal"
    >
      <div
        class="modal__dialog"
      >
        <div
          class="modal__header"
          role="presentation"
        >
          <div
            class="modal__icon"
          />
        </div>
        <p
          class="modal__title"
        />
        <div
          class="modal__body"
        >
          <div
            class="shipping-modal"
          >
            <div
              class="disclaimer"
            >
              <p
                class="shipping-modal__title"
              >
                Disclaimer
              </p>
              <p
                class="shipping-modal__title"
              >
                The price generated is a general quote.
              </p>
              <p
                class="shipping-modal__title"
              >
                You will receive an email confirmation with the final price after the Hemster team receives and reviews your garments and alteration requests.
              </p>
              <button
                class="btn btn__basic shipping-modal__btn btn__small btn__success"
                type="button"
              >
                Accept
              </button>
              <div
                class="disclaimer__btnWrapper"
              >
                <button
                  class="btn btn__basic disclaimer__btn btn__small btn__primary"
                  type="button"
                >
                  Decline
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal__backdrop"
      role="presentation"
    />
  </div>
</div>
`;
